src/AWSMetadataUtilities.jl

```diff
@@ -0,0 +1,372 @@
+    elseif protocol == "json"
+        json_version = service["jsonVersion"]
+        target = service["targetPrefix"]
+        return "const $service_id = AWS.JSONService(\"$service_name\", \"$api_version\", \"$json_version\", \"$target\")"
+    else
+        throw(ProtocolNotDefined("$service_id is using a new protocol; $protocol which is not supported."))
+    end
+end
+
+"""
+    _documentation_cleaning(documentation::String)
+
+Clean up the documentation to make it Julia compiler and human-readable.
+
+- Remove anything in-between <> HTML tags
+- Remove any dollar signs
+- Remove any \\
+
+# Arguments
+- `documentation::String`: Documentation to be cleaned
+
+# Returns
+- `String`: Cleaned documentation string
+"""
+function _documentation_cleaning(documentation::String)
+    documentation = replace(documentation, r"\<.*?\>" => "")
+    documentation = replace(documentation, '$' => ' ')
+    documentation = replace(documentation, '\\' => ' ')
+
+    return documentation
+end
+
+"""
+    _get_function_parameters(input, shapes)
+
+Get the required and optional parameters for a given operation.
+
+# Arguments
+- `input::String`: The input object for the given operation
+- `shapes::Dict{String, Any}`: Dictionary of all shapes for this AWS Service
+
+# Returns
+- `Tuple(Dict, Dict)`: (required_parameters, optional_parameters)
+"""
+function _get_function_parameters(input::String, shapes::Dict)
+    required_parameters = Dict{String, Any}()
+    optional_parameters = Dict{String, Any}()
+
+    input_shape = shapes[input]
+
+    if haskey(input_shape, "required")
+        for parameter in input_shape["required"]
+            required_parameters[parameter] = _documentation_cleaning(get(input_shape["members"][parameter], "documentation", ""))
+        end
+    end
+
+    if haskey(input_shape, "members")
+        for parameter in input_shape["members"]
+            parameter_name = parameter[1]
+
+            if !haskey(required_parameters, parameter_name)
+                optional_parameters[parameter_name] = _documentation_cleaning(get(parameter[2], "documentation", ""))
+            end
+        end
+    end
+
+    return (required_parameters, optional_parameters)
+end
+
+"""
+    _generate_high_level_definitions(
+        service_name::String,
+        protocol::String,
+        operations::Dict{String, Any},
+        shapes::Dict{String, Any}
+    )
+
+Generate high-level definitions for the `service`.
+
+# Arguments
+- `service_name::String`: AWS `service`
+- `protocol::String`: API protocol type
+- `operations::Dict{String, Any}`: Service API operations
+- `shapes::Dict{String, Any}`: All input shapes for this service
+
+# Return
+- `Array`: Array of all high-level definitions and documentation to be written into `services/{Service}.jl`
+"""
+function _generate_high_level_definitions(
+    service_name::String,
+    protocol::String,
+    operations::Dict,
+    shapes::Dict
+)
+    operation_definitions = []
```

Comment:
Minor performance improvement:
```suggestion
    operation_definitions = String[]
```